### PR TITLE
HPCC4J-489 DFSClient: fails to finish reading datasets in containeriz…

### PR DESCRIPTION
**Pull request by jpmcmu**
…ed environments

- Created workaround for Java socket stream available()==0 despite data being available.

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [X] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 1]
```diff
@@ -914,6 +914,13 @@ private void readDataInFetch()
             try
             {
                 bytesToRead = this.dis.available();
+
+                // Either due to a bug in the JVM or due to a design issue
+                // available can return 0 even in cases where there is data available.
+                // So always attempt to read at least some data
+                if (bytesToRead < MIN_SOCKET_READ_SIZE) {
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:921]: There seems to be many questions about his online 
(https://stackoverflow.com/questions/11982693/available-of-datainputstream-from-socket
https://stackoverflow.com/questions/26527918/datainputstream-available-returns-0-when-there-should-be-data-available)
It seems reading even if no data available is the accepted answer. 
I wonder if L921 should ensure available > 0 ( I think -1 is possible and signifies end of stream)

Also, curly brace should be in new line.

**Review by rpastrana — CHANGES_REQUESTED** [review]
@jpmcmu one comment

**Comment by jpmcmu** [comment]
@rpastrana updated issue# and implemented code review changes. Please review

**Review by rpastrana — APPROVED** [review]

**Pull request merged**